src/QLContext.actor.cpp

```diff
@@ -410,6 +415,14 @@ struct CompoundIndexPlugin : IndexPlugin, ReferenceCounted<CompoundIndexPlugin>,
 				for (int i = 0; i < nvv.size(); i++)
 					new_key.append(nvv[i].encode_key_part());
 				new_key.append(documentPath[documentPath.size() - 1]);
+				if (new_key.byteSize() > DocLayerConstants::INDEX_KEY_LENGTH_LIMIT) {
+					TraceEvent(SevError, "CompoundIndexKeyTooLarge")
+					    .detail("Offending Key size", new_key.byteSize())
```

Comment:
`TraceEvent` detail keys are usually a single word, with no spaces. As they will convert into xml keys and then log system can use them to index. This could become `OffendingKeySize` or `KeySize`. Same for `Index Name`